ush/python/pygfs/task/snowens_analysis.py

```diff
@@ -0,0 +1,371 @@
+
+        for bkg_time in bkg_times:
+            for mem in range(1, self.task_config.NMEM_ENS + 1):
+                # for now, just looping serially, should parallelize this eventually
+                logger.info(f"Now applying increment to member mem{mem:03}")
+                logger.info(f'{os.path.join(self.task_config.DATA, "anl", f"mem{mem:03}")}')
+                memdict = AttrDict(
+                    {
+                        'HOMEgfs': self.task_config.HOMEgfs,
+                        'DATA': os.path.join(self.task_config.DATA, "anl", f"mem{mem:03}"),
+                        'DATAROOT': self.task_config.DATA,
+                        'current_cycle': bkg_time,
+                        'CASE_ENS': self.task_config.CASE_ENS,
+                        'OCNRES': self.task_config.OCNRES,
+                        'ntiles': 6,
+                        'ENS_APPLY_INCR_NML_TMPL': self.task_config.ENS_APPLY_INCR_NML_TMPL,
+                        'APPLY_INCR_EXE': self.task_config.APPLY_INCR_EXE,
+                        'APRUN_APPLY_INCR': self.task_config.APRUN_APPLY_INCR,
+                        'MYMEM': f"{mem:03}",
+                    }
+                )
+                self.add_increments(memdict)
+
+    @staticmethod
+    @logit(logger)
+    def get_bkg_dict(config: Dict) -> Dict[str, List[str]]:
+        """Compile a dictionary of model background files to copy
+
+        This method constructs a dictionary of FV3 RESTART files (coupler, sfc_data)
+        that are needed for global snow DA and returns said dictionary for use by the FileHandler class.
+
+        Parameters
+        ----------
+        config: Dict
+            Dictionary of key-value pairs needed in this method
+            Should contain the following keys:
+            COMIN_ATMOS_RESTART_PREV
+            DATA
+            current_cycle
+            ntiles
+
+        Returns
+        ----------
+        bkg_dict: Dict
+            a dictionary containing the list of model background files to copy for FileHandler
+        """
+
+        bkg_dict = {
+            'mkdir': [],
+            'copy': [],
+        }
+        return bkg_dict
+
+    @staticmethod
+    @logit(logger)
+    def add_increments(config: Dict) -> None:
+        """Executes the program "apply_incr.exe" to create analysis "sfc_data" files by adding increments to backgrounds
+
+        Parameters
+        ----------
+         config: Dict
+             Dictionary of key-value pairs needed in this method
+             Should contain the following keys:
+             HOMEgfs
+             DATA
+             DATAROOT
+             current_cycle
+             CASE
+             OCNRES
+             ntiles
+             APPLY_INCR_NML_TMPL
+             APPLY_INCR_EXE
+             APRUN_APPLY_INCR
+
+        Raises
+        ------
+        OSError
+            Failure due to OS issues
+        WorkflowException
+            All other exceptions
+        """
+        os.chdir(config.DATA)
```

Comment:
Using `os.chdir` here.  Seen `wxflow` `chdir` elsewhere in this script.  Just FYI.

Reply:
I don't remember the details, but there was a reason why I did this. For whatever reason, `wxflow chdir` was not working and it didn't `cd` into each subdirectory, but this did work.